doc/specs/#3327 - Application Theming/#3327 - Application Theming.md

```diff
@@ -0,0 +1,500 @@
+picker "clears" the color it has set for the tab, it should revert to the color
+from the theme (if one is set).
+
+Also mentioned in the implementation of the color picker feature was the ability
+to not set the entire color of the tab, but just the color of a tab "overline",
+similar to the way Firefox (by default) styles the focused tab.
+
+Currently, the `TabView` doesn't support a tab "overline" like this, however, in
+the future where this is possible, we'd love to also support such an overline.
+However, the story of setting the tab color manually becomes a bit more
+confusing now.
+
+* The user should be able to set both the `tab.background` and `tab.overline`
+  colors in a theme.
+* The user should be able to configure whether the color picker sets the
+  `background` or the `overline` color of the tab.
+
+The second setting added above will allow the user to change what's controlled
+by the color picker. Similarly to how the color picker can set the background of
+the tab to override the background from the theme, the user could configure the
+color picker to be able to change the overline color, not the background color
+of the tab. Then, when the user uses the color picker, the overline color will
+be overridden by the color picker, instead of the tab background color.
+
+**Other things to consider:**
+* Users might want to be able to set a tab color as a part of the profile. One
+  could imagine wanting to set the tab background color for Windows PowerShell
+  to `rgb(1, 36, 86)` automatically. If we make this property part of the
+  Profile, then we should use the profile's value as the runtime-override (of
+  the theme value) for this property. If the color picker is used to set the
+  color of the tab, then it'll override the runtime color for that tab.
+  - How does this interact with multiple Panes in a tab? Should the Tab
+    override color be per-terminal instance? If the terminal has a tab color,
+    then that would override the theme, but not the tab's own override color?
+    - If that were the case, the order of precedence would be:
+      1. A color set at runtime with the color picker
+      2. A color from the active terminal within the tab, if it has one
+      3. The tab color from the theme
+* Users might want to be able to configure the titlebar to use a color based off
+  the active tab color. We might need an additional special value like
+  `terminalBackground` that lets users say "I want to use the active tab color
+  as the titlebar color".
+  - During [#3789], there was a point where the terminal raised actually
+    implemented something like this. In it's implementation, the titlebar color
+    would be slightly lighter or darker than the tab color (to provide some
+    contrast). We'd want to make sure that the user could specify both "I want
+    to use the tab color with some contrast applied" or just literally "Use
+    whatever the active tab's color is."
+
+### Default Themes

```

Comment:
(Maybe worth mentioning in future considerations over v1) Should we consider a "defaults/list" structure like we did with `profiles.defaults`? For now, I'm guessing themes won't be too big, but over time that could become an issue (maybe)

Reply:
See, I don't think we need a "defaults" for themes. People will only ever have one theme active at a time, for the entire window, so we don't really need to abstract that away. 

The only reason I've got a whole list of themes is so: 
* In the future, people can use extensions to install multiple themes, and just pick one they like
* People can iterate on ideas they like, and keep around old "themes" they create, even if they're using something else currently. 

I think it's fine for each theme to be it's own atomic object, not composed of a (user theme defaults)+(changes to the user theme defaults)